make/protogen.mk

```diff
@@ -108,10 +108,12 @@ ROX_M_ARGS = $(foreach proto,$(ALL_PROTOS_REL),M$(proto)=github.com/stackrox/rox
 SCANNER_M_ARGS = $(foreach proto,$(ALL_SCANNER_PROTOS_REL),M$(proto)=github.com/stackrox/scanner/generated/$(patsubst %/,%,$(dir $(proto))))
 # Combine the *_M_ARGS.
 M_ARGS = $(ROX_M_ARGS) $(SCANNER_M_ARGS)
-# This is the M_ARGS used for the grpc-gateway invocation. We only map the storage protos, because
+# This is the M_ARGS used for the grpc-gateway invocation. We only map the storage and api/v1 protos, because
 # - the gateway code produces no output (possibly because of a bug) if we pass M_ARGS_STR to it.
-# - the gateway code doesn't need access to anything outside api/v1 except storage. In particular, it should NOT import internalapi protos.
-GATEWAY_M_ARGS = $(foreach proto,$(STORAGE_PROTOS),M$(proto)=github.com/stackrox/rox/generated/$(patsubst %/,%,$(dir $(proto))))
+# - the gateway code only needs access to api/v1, api/v2 and storage/. In particular, it should NOT import internalapi protos.
+GATEWAY_M_ARGS := $(foreach proto,$(STORAGE_PROTOS),M$(proto)=github.com/stackrox/rox/generated/$(patsubst %/,%,$(dir $(proto))))
+GATEWAY_M_ARGS += $(foreach proto,$(API_SERVICE_PROTOS),M$(proto)=github.com/stackrox/rox/generated/$(patsubst %/,%,$(dir $(proto))))
```

Comment:
`ln14` `API_SERVICE_PROTOS` does not filter on v2

Reply:
We don't need it to filter on v2 unless v1 ports also need to use v2 types. But I thought it would always be v2 protos having to use some common v1 types.

Reply:
With the transition to v2, it should have it's own base protos. Let us not import anything from v1 into v2.

Reply:
Replicating v1.Empty and v1.ResourceByID is easy, but replicating everything from search_service.proto would result in a much bigger change. I think we would at least need to replicate v1.RawQuery and the proto messages needed for Autocomplete search and global search.

Reply:
Yes, the proposed approach is reasonable one. Strictly we do not want to import `api/v1` into `api/v2`. This is because changing definitions in v1 for the purposes of v1 will impact v2. Similarly, changing imported v1 for the purposes of v2 will impact it elsewhere in v1 when the required change is only meant for v2.